pkg/config/environment_detection.go

```diff
@@ -114,7 +114,9 @@ func DetectFeatures() {
 			}
 		}
 
-		log.Infof("Features detected from environment: %v", newFeatures)
+		if len(newFeatures) > 0 {
+			log.Infof("Features detected from environment: %v", newFeatures)
```

Comment:
IMO, even if the `newFeature` slice is empty it can be good to keep this log line.

maybe
```suggestion
			log.Infof("%d Features detected from environment: %v", len(newFeatures), newFeatures)
```

Reply:
👍